### ISSUE TITLE
Batch Normalization의 효과는?

### ISSUE THREAD

**Issue by import-max** [AInsteins/AI-Tech-Interview#47]

**Comment by kimyuuuuuu** [comment]
- 배치 정규화(Batch Normalization)
    - 학습 시 **미니배치 단위로 입력의 분포가 평균이 0, 분산이 1이 되도록 정규화**한다.
    
`장점 1` 기울기 소실/폭발 문제가 해결되어 큰 학습률을 설정할 수 있어 학습속도가 빨라진다.

`장점 2` 항상 입력을 정규화시키기 때문에 가중치 초깃값에 크게 의존하지 않아도 된다.

`장점 3` 자체적인 규제(Regularization) 효과가 있어 Dropout이나 Weight Decay와 같은 규제 방법을 사용하지 않아도 된다.

**Comment by kimyuuuuuu** [comment]
<질문>
- 왜 자체적인 규제 효과가 있는가?
- 어떻게 학습이랑 추론이 되는가..?

**Comment by miinkang** [comment]
- batch normalization
    - 학습시 미니배치 단위로 입력 분포의 평균이 0, 분산이 1이 되도록 정규화하는 것
    - 학습 과정 자체를 안정화하는 역할
- 효과
    - 기울기 소실/폭발 문제가 해결되어 큰 학습률을 설정할 수 있고, 학습 속도가 빨라질 수 있다.
    - 가중치 초깃값에 크게 의존하지 않아도 된다.
    - regularization 효과와 같기 때문에 dropout을 쓰지 않아도 된다.
        - Mini-batch에 들어가는 평균과 분산이 지속적으로 변화는 과정 속에서 분포가 조금씩 바뀌게 되고 학습하는 과정에 weight에 영향을 주게 되기 때문
- 추론 단계에서 batch normalization
    - 추론에서는 평균, 분산을 구하기 어렵기 때문에 학습 단계의 평균, 분산을 저장한 후 사용한다.
        - 이는 이동평균이나 지수평균을 이용해 계산한 것

**Comment by jarammm** [comment]
# batch normalization
- 학습 과정에서 계층 별로 입력의 데이터 분포가 달라지는 현상인 Internal Covariant Shift를 이겨내기 위해 나온 방법이 batch normalization
- 다양한 분포를 가지더라도 **각 배치별로 평균과 분산을 이용해 정규화**
- 정규화를 통하여 분포를 zero mean gaussian 형태(표준정규분포 따름)로 만듦

# batch normalization의 효과
- weight initialization에 덜 민감
- learning rate의 크기를 키우는 데에 좀 더 과감한 시도가 가능
- Batch Normalization에서 평균과 분산이 지속적으로 변하고 weight 업데이트에도 계속 영향을 주어 한 weight에 가중치가 큰 방향으로만 학습되지 않기 때문에 Regularization effect 얻을 수 있음 -> overfitting 문제에 좀 더 강건(dropout이 필수가 아니게 됨)
-
